### DISCUSSION TITLE
Single correction factor for all the cells/anchors when integrating query with reference

### DISCUSSION THREAD

**Discussion by aminzia** [satijalab/seurat#4480] [Ideas]
Dear Satijlab, 

I have a general question regarding your anchor-based integration. 

The method implemented is sophisticated in the sense that it computes different correction factors for different clusters/cell types in the query and reference samples. 

However, it seems that different correction factors for different anchors (and therefore cell types) within a sample will change the relative expression of markers of the cell types within that sample. In fact, by aligning cell types individually between samples, we sacrifice the accuracy of within-sample expression patterns for between-sample cell-type marker alignment. This makes sense if we only look for changes, e.g. in the same cell types, between two samples. But does not seem to make sense for cases where the comparison between cell types is important after integration (i.e. cluster marker identification). 

One needs to ask why there is a batch effect in the first place? It seems to me that we only have one batch bias for all the cell types within a sample, not different biases for each cluster/cell-type. So unless each cell type has a different bias, I think we need only one correction factor (perhaps averaged over all anchors) to preserve the relative expression of cell types within each sample (e.g. a single rotation operation to keep intra-sample variability unchanged which to me is as important as between-sample variability).

So, my main question is that whether there is a way in your functions to set all the correction factors to be unified/combined/averaged for all the anchors before integration? 

Thank you
Amin

**Comment by timoast** [comment]
> However, it seems that different correction factors for different anchors (and therefore cell types) within a sample will change the relative expression of markers of the cell types within that sample. In fact, by aligning cell types individually between samples, we sacrifice the accuracy of within-sample expression patterns for between-sample cell-type marker alignment. This makes sense if we only look for changes, e.g. in the same cell types, between two samples. But does not seem to make sense for cases where the comparison between cell types is important after integration (i.e. cluster marker identification).

This is one of the reasons why we always recommend using the uncorrected data for differential expression

>  It seems to me that we only have one batch bias for all the cell types within a sample, not different biases for each cluster/cell-type.

This is not typically true

> So, my main question is that whether there is a way in your functions to set all the correction factors to be unified/combined/averaged for all the anchors before integration?

If you wanted to do this, you should use bulk-cell batch correction tools like ComBat or limma. It was shown in https://www.nature.com/articles/nbt.4096 that this approach typically does not perform well on single-cell data
